src/pybind/mgr/mds_autoscaler/module.py

```diff
@@ -0,0 +1,84 @@
+"""
+Automatically scale MDSs based on status of the file-system using the FSMap
+"""
+
+import errno
+import json
+import time
+import mgr_util
+import threading
+import uuid
+import logging
+from six import itervalues, iteritems
+from collections import defaultdict
+from prettytable import PrettyTable, PLAIN_COLUMNS
+from mgr_module import MgrModule
+
+import orchestrator
+
+log = logging.getLogger(__name__)
+
+"""
+Some terminology is made up for the purposes of this module:
+
+ - "grow/shrink" - increase/decrease the MDSs in the system
+"""
+
+class MdsAutoscaler(MgrModule):
+    """
+    MDS autoscaler.
+    """
+    NATIVE_OPTIONS = [
+        'max_mds'
+    ]
+
+    MODULE_OPTIONS = [
+        {
+            'name': 'sleep_interval',
+            'default': str(60),
+        },
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super(MdsAutoscaler, self).__init__(*args, **kwargs)
+        self.fs_map = None
+
+    def create_mds(self, fs_name):
+        spec = orchestrator.StatelessServiceSpec(fs_name)
+        try:
+            completion = self.add_mds(spec)
```

Comment:
If we want this to acutally work, we need to pass through a placmenet spec. Otherwise `cephadm` won't be able to deploy it.

Reply:
with #33808 it will work with no placement at all